Makefile

```diff
@@ -29,7 +29,7 @@ endif
 CFLAGS_EXTRA ?=
 CXXFLAGS_EXTRA ?=
 CFLAGS=-O3 -fno-exceptions $(CFLAGS_EXTRA)
-CXXFLAGS=-O3 -fno-exceptions -fno-omit-frame-pointer -fvisibility=hidden -std=c++11 $(CXXFLAGS_EXTRA)
+CXXFLAGS=-O3 -fno-exceptions -fno-omit-frame-pointer -fno-optimize-sibling-calls -fvisibility=hidden -std=c++11 $(CXXFLAGS_EXTRA)
```

Comment:
This is needed to preserve `calloc_hook_dummy` from being optimized away in `-O2+`

Reply:
You essentially disable a useful optimization for the whole profiler on all platforms just to satisfy a couple of tests on musl. Doesn't look right to me. Can you fine-tune optimizations for the particular function instead? (using compiler-specific attributes)

Reply:
2abe0fdb19e92248fa39061e4936c4a6f09a18c8